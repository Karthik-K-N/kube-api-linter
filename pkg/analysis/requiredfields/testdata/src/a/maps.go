package a

type TestMaps struct {
	// +required
	Map map[string]string `json:"map"` // want "field Map should have the omitempty tag."

	// +required
	MapWithOmitEmpty map[string]string `json:"mapWithOmitEmpty,omitempty"`

	// +required
	MapPtr *map[string]string `json:"mapPtr"` // want "field MapPtr should have the omitempty tag." "field MapPtr underlying type does not need to be a pointer. The pointer should be removed."

	// +required
	MapPtrWithOmitEmpty *map[string]string `json:"mapPtrWithOmitEmpty,omitempty"` // want "field MapPtrWithOmitEmpty underlying type does not need to be a pointer. The pointer should be removed."

	// +kubebuilder:validation:MinProperties=1
	// +required
	MapWithPositiveMinProperties map[string]string `json:"mapWithPositiveMinProperties"` // want "field MapWithPositiveMinProperties should have the omitempty tag."

	// +kubebuilder:validation:MinProperties=1
	// +required
	MapWithPositiveMinPropertiesWithOmitEmpty map[string]string `json:"mapWithPositiveMinPropertiesWithOmitEmpty,omitempty"`

	// +kubebuilder:validation:MinProperties=0
	// +required
	MapWithZeroMinProperties map[string]string `json:"mapWithZeroMinProperties"` // want "field MapWithZeroMinProperties should have the omitempty tag."

	// +kubebuilder:validation:MinProperties=0
	// +required
	MapWithZeroMinPropertiesWithOmitEmpty map[string]string `json:"mapWithZeroMinPropertiesWithOmitEmpty,omitempty"`
}
