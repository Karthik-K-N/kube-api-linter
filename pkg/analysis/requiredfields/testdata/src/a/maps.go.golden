package a

type TestMaps struct {
    // +required
	Map map[string]string `json:"map,omitempty"` // want "field Map should have the omitempty tag."

    // +required
	MapWithOmitEmpty map[string]string `json:"mapWithOmitEmpty,omitempty"`

    // +required
	MapPtr map[string]string `json:"mapPtr,omitempty"` // want "field MapPtr should have the omitempty tag." "field MapPtr underlying type does not need to be a pointer. The pointer should be removed."

    // +required
	MapPtrWithOmitEmpty map[string]string `json:"mapPtrWithOmitEmpty,omitempty"` // want "field MapPtrWithOmitEmpty underlying type does not need to be a pointer. The pointer should be removed."

	// +kubebuilder:validation:MinProperties=1
	// +required
    MapWithPositiveMinProperties map[string]string `json:"mapWithPositiveMinProperties,omitempty"` // want "field MapWithPositiveMinProperties should have the omitempty tag."

	// +kubebuilder:validation:MinProperties=1
	// +required
    MapWithPositiveMinPropertiesWithOmitEmpty map[string]string `json:"mapWithPositiveMinPropertiesWithOmitEmpty,omitempty"`

	// +kubebuilder:validation:MinProperties=0
	// +required
    MapWithZeroMinProperties map[string]string `json:"mapWithZeroMinProperties,omitempty"` // want "field MapWithZeroMinProperties should have the omitempty tag."

	// +kubebuilder:validation:MinProperties=0
	// +required
    MapWithZeroMinPropertiesWithOmitEmpty map[string]string `json:"mapWithZeroMinPropertiesWithOmitEmpty,omitempty"`
}
